tests/test_build.py

```diff
@@ -51,3 +51,16 @@ def test_pickle(pickle_protocol: int) -> None:
     v2 = pickle.loads(p)
     assert v2.validate_python('tRuE') is True
     assert repr(v1) == repr(v2)
+
+
+def test_schema_recursive_error():
+    schema = {'type': 'union', 'choices': []}
+    schema['choices'].append(schema)
+    with pytest.raises(SchemaError, match='Recursive detected, depth exceeded max allowed value of 100'):
```

Comment:
Wouldn't it make sense for this to be a `RecursionError`?

Reply:
Changing it will make the error more readable hopefully.

Reply:
It's be very keen on that if there's a way in rust, but I couldn't find anything.

I got we'd need to call the standard python methods.